tests/components/whirlpool/test_climate.py

```diff
@@ -0,0 +1,364 @@
+"""Test the Whirlpool Sixth Sense climate domain."""
+from unittest.mock import AsyncMock, MagicMock
+
+import aiohttp
+import whirlpool
+
+from homeassistant.components.climate.const import (
+    ATTR_CURRENT_HUMIDITY,
+    ATTR_CURRENT_TEMPERATURE,
+    ATTR_FAN_MODE,
+    ATTR_FAN_MODES,
+    ATTR_HVAC_MODE,
+    ATTR_HVAC_MODES,
+    ATTR_MAX_TEMP,
+    ATTR_MIN_TEMP,
+    ATTR_SWING_MODE,
+    ATTR_SWING_MODES,
+    ATTR_TARGET_TEMP_STEP,
+    DOMAIN as CLIMATE_DOMAIN,
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_MIDDLE,
+    FAN_OFF,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_COOL,
+    HVAC_MODE_DRY,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SERVICE_SET_FAN_MODE,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_SWING_MODE,
+    SERVICE_SET_TEMPERATURE,
+    SUPPORT_FAN_MODE,
+    SUPPORT_SWING_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+    SWING_HORIZONTAL,
+    SWING_OFF,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_FRIENDLY_NAME,
+    ATTR_SUPPORTED_FEATURES,
+    ATTR_TEMPERATURE,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from . import init_integration
+
+
+async def update_ac_state(hass: HomeAssistant, mock_aircon_api: MagicMock):
+    """Simulate an update trigger from the API."""
+    schedule_update_ha_state_cb = mock_aircon_api.call_args.args[2]
+    schedule_update_ha_state_cb()
+    await hass.async_block_till_done()
+    return hass.states.get("climate.said1")
+
+
+async def test_no_appliances(hass: HomeAssistant, mock_auth_api: MagicMock):
+    """Test the setup of the climate entities when there are no appliances available."""
+    mock_auth_api.return_value.get_said_list.return_value = []
+    await init_integration(hass)
+    assert len(hass.states.async_all()) == 0
+
+
+async def test_name_fallback_on_exception(
+    hass: HomeAssistant, mock_aircon_api: MagicMock
+):
+    """Test name property."""
+    mock_aircon_api.return_value.fetch_name = AsyncMock(
+        side_effect=aiohttp.ClientError()
+    )
+
+    await init_integration(hass)
+    state = hass.states.get("climate.said1")
+    assert state.attributes.get(ATTR_FRIENDLY_NAME) == "said1"
+
+
+async def test_static_attributes(hass: HomeAssistant, mock_aircon_api: MagicMock):
+    """Test static climate attributes."""
+    await init_integration(hass)
+
+    entry = er.async_get(hass).async_get("climate.said1")
+    assert entry
+    assert entry.unique_id == "said1"
+
+    state = hass.states.get("climate.said1")
+    assert state is not None
+    assert state.state != STATE_UNAVAILABLE
+    assert state.state == HVAC_MODE_COOL
+
+    attributes = state.attributes
+    assert attributes.get(ATTR_FRIENDLY_NAME) == "TestZone"
```

Comment:
```suggestion
    assert attributes[ATTR_FRIENDLY_NAME] == "TestZone"
```

Use `[]` when we except the key to be there. Only use `.get()` when we don't know if its going to be present

Reply:
done